resolver/resolver.go

```diff
@@ -2090,6 +2136,25 @@ func (r *Resolver) resolveMessageCELValues(ctx *context, env *cel.Env, msg *Mess
 			}
 		}
 	}
+	for valIdx, validation := range msg.Rule.Validations {
+		if err := r.resolveCELValue(ctx, env, validation.Error.ValidationRule); err != nil {
+			ctx.addError(
+				ErrWithLocation(
+					err.Error(),
+					source.MessageValidationLocation(msg.File.Name, msg.Name, valIdx),
```

Comment:
The error location to be in the validation rule field, not in the validation itself.